libcudacxx/include/cuda/std/__bit/byteswap.h

```diff
@@ -0,0 +1,121 @@
+         | (static_cast<_Full>(__impl(static_cast<_Half>(__val))) << CHAR_BIT * sizeof(_Half));
+  }
+
+public:
+  template <class _Tp>
+  _CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI static _CCCL_CONSTEXPR_CXX14 _Tp __impl(_Tp __val) noexcept
+  {
+    _Tp __result{};
+    for (size_t __i{}; __i < sizeof(__val); ++__i)
+    {
+      __result <<= CHAR_BIT;
+      __result |= (__val >> (__i * CHAR_BIT)) & static_cast<_Tp>(UCHAR_MAX);
+    }
+    return __result;
+  }
+
+  _CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI static constexpr uint16_t __impl(uint16_t __val) noexcept
+  {
+#if defined(_CCCL_BUILTIN_BSWAP16)
+    return _CCCL_BUILTIN_BSWAP16(__val);
+#else // ^^^ _CCCL_BUILTIN_BSWAP16 ^^^ / vvv !_CCCL_BUILTIN_BSWAP16 vvv
+    return (__val << CHAR_BIT) | (__val >> CHAR_BIT);
+#endif // !_CCCL_BUILTIN_BSWAP16
+  }
+
+  _CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI static constexpr uint32_t __impl(uint32_t __val) noexcept
+  {
+#if defined(_CCCL_BUILTIN_BSWAP32)
+    return _CCCL_BUILTIN_BSWAP32(__val);
+#else // ^^^ _CCCL_BUILTIN_BSWAP32 ^^^ / vvv !_CCCL_BUILTIN_BSWAP32 vvv
+    return __impl_recursive<uint16_t>(__val);
+#endif // !_CCCL_BUILTIN_BSWAP32
+  }
+
+  _CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI static constexpr uint64_t __impl(uint64_t __val) noexcept
+  {
+#if defined(_CCCL_BUILTIN_BSWAP64)
+    return _CCCL_BUILTIN_BSWAP64(__val);
+#else // ^^^ _CCCL_BUILTIN_BSWAP64 ^^^ / vvv !_CCCL_BUILTIN_BSWAP64 vvv
+    return __impl_recursive<uint32_t>(__val);
+#endif // !_CCCL_BUILTIN_BSWAP64
+  }
+
+#if !defined(_LIBCUDACXX_HAS_NO_INT128)
+  _CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI static constexpr unsigned __int128 __impl(unsigned __int128 __val) noexcept
+  {
+#  if defined(_CCCL_BUILTIN_BSWAP128)
+    return _CCCL_BUILTIN_BSWAP128(__val);
+#  else // ^^^ _CCCL_BUILTIN_BSWAP128 ^^^ / vvv !_CCCL_BUILTIN_BSWAP128 vvv
+    return __impl_recursive<uint64_t>(__val);
+#  endif // !_CCCL_BUILTIN_BSWAP128
+  }
+#endif // !_LIBCUDACXX_HAS_NO_INT128
+};
+
+_CCCL_TEMPLATE(class _Integer)
+_CCCL_REQUIRES(_CCCL_TRAIT(is_integral, _Integer) _CCCL_AND(sizeof(_Integer) == 1))
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI constexpr _Integer byteswap(_Integer __val) noexcept
+{
+  return __val;
+}
+
+_CCCL_TEMPLATE(class _Integer)
+_CCCL_REQUIRES(_CCCL_TRAIT(is_integral, _Integer) _CCCL_AND(sizeof(_Integer) > 1)
+                 _CCCL_AND(has_single_bit(sizeof(_Integer))))
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI constexpr _Integer byteswap(_Integer __val) noexcept
+{
+  return static_cast<_Integer>(__byteswap_impl::__impl(__to_unsigned_like(__val)));
+}
+
+_CCCL_TEMPLATE(class _Integer)
+_CCCL_REQUIRES(_CCCL_TRAIT(is_integral, _Integer) _CCCL_AND(sizeof(_Integer) > 1)
+                 _CCCL_AND(!has_single_bit(sizeof(_Integer))))
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI _CCCL_CONSTEXPR_CXX14 _Integer byteswap(_Integer __val) noexcept
+{
+  return static_cast<_Integer>(__byteswap_impl::__impl(__to_unsigned_like(__val)));
```

Comment:
Sorry I missed it, but we need to fully qualify the function call here
```suggestion
  return static_cast<_Integer>(__byteswap_impl::__impl(_CUDA_VSTD::__to_unsigned_like(__val)));
```